internal/agent/agent.go

```diff
@@ -148,6 +148,22 @@ func (a *Agent) Run(ctx context.Context) error {
 			}
 			systemInfoManager.RegisterCollector(ctx, "tpmVendorInfo", tpmClient.TpmVendorInfoCollector)
 			systemInfoManager.RegisterCollector(ctx, "attestation", tpmClient.TpmAttestationCollector)
+
+			a.log.Warn("Experimental features enabled: creating TPM hardware identity")
+
+			// overwrite previous CSR and device name
+			tpmSigner := tpmClient.GetLDevIDSigner()
+			publicKey := tpmClient.GetLDevIDPublic()
+			publicKeyHash, err := fcrypto.HashPublicKey(publicKey)
+			if err != nil {
+				a.log.Errorf("Unable to create public key hash for TPM key: %v", err)
```

Comment:
do we want to return here? if we can't generate the hash I assume `publicKeyHash` is not usable?

Reply:
Yeah, I added some logic to go ahead and create SW keys in this case